[Priest] - Disc implementation for Damage

dj-riff opened the pull request:
Added Power of the Dark Side proc and Sins of the Many talent. APL still needs to be optimized. Schism debuff needs to be implemented as well (fairly certain it doesn't work).

dj-riff committed:
Added Disc Implementation and fixed some spelling errors

Added Power of the Dark Side and Sins of the Many to disc module. Fixed minor spelling errors in shadow module.

dj-riff committed:
Added stuff from recent pushes

Added stuff from recent pushes

dj-riff committed:
[Priest] - Fixed bool error in spell ready

[Priest] - Fixed bool error in spell ready

seanpeters86 commented on engine/class_modules/priest/sc_priest.cpp:
```diff
@@ -803,6 +803,8 @@ void priest_t::create_procs()
   procs.serendipity              = get_proc( "Serendipity (Non-Tier 17 4pc)" );
   procs.serendipity_overflow     = get_proc( "Serendipity lost to overflow (Non-Tier 17 4pc)" );
+  procs.power_of_the_dark_side   = get_proc("Power of the Dark Side Penance damage buffed");
+  procs.power_of_the_dark_side_overflow = get_proc("Power of the Dark Side lost to overflow");
```
can you line up the `=`'s here? so

```
  procs.surge_of_light_overflow         = get_proc( "Surge of Light lost to overflow" );
  procs.serendipity                     = get_proc( "Serendipity (Non-Tier 17 4pc)" );
  procs.serendipity_overflow            = get_proc( "Serendipity lost to overflow (Non-Tier 17 4pc)" );
  procs.power_of_the_dark_side          = get_proc( "Power of the Dark Side Penance damage buffed" );
  procs.power_of_the_dark_side_overflow = get_proc( "Power of the Dark Side lost to overflow" );
``` 
and then add spaces between the parens and quotes

seanpeters86 commented on engine/class_modules/priest/sc_priest.cpp:
```diff
@@ -1188,7 +1190,7 @@ void priest_t::init_spells()
 
   // Mastery Spells
-  mastery_spells.absolution    = find_mastery_spell( PRIEST_DISCIPLINE );
+  mastery_spells.grace    = find_mastery_spell( PRIEST_DISCIPLINE );
```
same thing about lining up `=`'s

seanpeters86 commented on engine/class_modules/priest/sc_priest.hpp:
```diff
@@ -986,6 +1016,10 @@ struct priest_action_t : public Base
+        {priest().specs.discipline_priest->effectN( 1 ),  affected_by.discipline_priest_da},
+        {priest().specs.discipline_priest->effectN( 2 ),  affected_by.discipline_priest_ta},
+        {priest().talents.sins_of_the_many->effectN( 1), affected_by.sins_of_the_many_da},
+        //{priest().buffs.sins_of_the_many->data().effectN( 1), affected_by.sins_of_the_many_ta },
```
can you add a comment for why this is still left in commented in? Just so that folks know why its not being used

seanpeters86 commented on engine/class_modules/priest/sc_priest_discipline.cpp:
```diff
@@ -270,6 +309,7 @@ void priest_t::init_spells_discipline()
   talents.dominant_mind = find_talent_spell( "Dominant Mind" );
   // T75
   talents.sanctuary       = find_talent_spell( "Sanctuary" );
+  talents.sins_of_the_many = find_talent_spell( "Sins of the Many" );
```
same thing about lining up `=`'s

seanpeters86 commented on engine/class_modules/priest/sc_priest_discipline.cpp:
```diff
@@ -295,6 +335,8 @@ void priest_t::init_spells_discipline()
   specs.spirit_shell    = find_specialization_spell( "Spirit Shell" );
   specs.enlightenment   = find_specialization_spell( "Enlightenment" );
+  specs.discipline_priest = find_specialization_spell( "Discipline Priest" );
+  specs.power_of_the_dark_side = find_spell( 198069 ); //Damage ID of Power of the Dark Side
```
same thing about lining up `=`'s

seanpeters86 commented on engine/class_modules/priest/sc_priest_holy.cpp:
```diff
@@ -230,7 +230,7 @@ void priest_t::generate_apl_holy_d()
 
     // Precombat actions
-    precombat->add_action("potion");
+	precombat->add_action("potion");
```
any reason this is indented?

seanpeters86 commented on engine/class_modules/priest/sc_priest_holy.cpp:
```diff
@@ -245,23 +245,38 @@ void priest_t::generate_apl_holy_d()
     default_list->add_action("ancestral_call,if=raid_event.adds.in>20|raid_event.adds.remains>10|raid_event.adds.duration<10");
-    default_list->add_talent(this, "Divine Star", "if=(raid_event.adds.in>5|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.divine_star>1" );
-    default_list->add_talent(this, "Halo", "if=(raid_event.adds.in>14|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.halo>0" );
+    default_list->add_talent(this, "Divine Star", "if=(raid_event.adds.in>5|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.divine_star>1");
```
i think technically the APL is faster if you put the `spell_targets.divine_star>1` at the first of the condition

lgkern replied:
correct, always put the most restrictive condition first, as that will happen more often

dj-riff replied:
I didn't actually mess with that at all, i was copying stuff julia had sent, so I'll just omit the holy file.

seanpeters86 commented on engine/class_modules/priest/sc_priest_holy.cpp:
```diff
@@ -245,23 +245,38 @@ void priest_t::generate_apl_holy_d()
-    default_list->add_talent(this, "Divine Star", "if=(raid_event.adds.in>5|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.divine_star>1" );
-    default_list->add_talent(this, "Halo", "if=(raid_event.adds.in>14|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.halo>0" );
+    default_list->add_talent(this, "Divine Star", "if=(raid_event.adds.in>5|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.divine_star>1");
+    default_list->add_talent(this, "Halo", "if=(raid_event.adds.in>14|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.halo>0");
```
same thing as above with speed of APL `spell_targets.halo>0` moving to the front

seanpeters86 commented on engine/class_modules/priest/sc_priest_holy.cpp:
```diff
@@ -245,23 +245,38 @@ void priest_t::generate_apl_holy_d()
     default_list->add_action(this, "Smite");
     default_list->add_action(this, "Holy Fire");
-    default_list->add_talent(this, "Divine Star", "if=(raid_event.adds.in>5|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.divine_star>0" );
+    default_list->add_talent(this, "Divine Star", "if=(raid_event.adds.in>5|raid_event.adds.remains>2|raid_event.adds.duration<2)&spell_targets.divine_star>0");
```
same thing as above

lgkern reviewed (commented):
All changes Publik already mentioned. 
Make sure you're using the latest Visual Studio 2017 version, as the simulationcraft project has a definition on how spaces and other formatting should be done and, by using the latest Visual Studio it automatically applies that layout to the code.

Also, discard the changes on the sc_priest_shadow.cpp as they only remove formatting.

dj-riff committed:
Fixing issues with Holy/Shadow modules and formatting

Fixing issues with Holy/Shadow modules and formatting

dj-riff committed:
[Priest] - Update fromatting

[Priest] - Update fromatting to match rest of modules

dj-riff commented:
I've pulled the shadow/holy files from the base simc and reuploaded them (somehow the formatting got messed up in my fork?). That addressed the issues with Holy/Shadow. I also fixed the formatting issues as well. I was still using VS2015, swapping to 2017 made my life a billion times easier. Ty ansh.
